File: dec1/two.mjs
import fs from "fs/promises";

(async function () {
    let file = await fs.readFile("./input.txt", "utf-8");
    let one = [];
    let two = [];
    file.split("\n").forEach(line => {
        let nums = line.split(" ");
        one.push(+nums[0]);
        two.push(+nums[3]);
    });
    let res = 0;
    one.forEach(num => {
        res += num * two.filter(el => el === num).length
    })
    console.log(res);
})()

File: dec1/two.rs
use std::fs;

fn main () {
    let file = fs::read_to_string("input.txt").unwrap();
    let mut one = Vec::new();
    let mut two = Vec::new();
    for line in file.lines() {
        let arr: Vec<&str> = line.split_whitespace().collect();
        one.push(arr[0].parse::<i32>().unwrap());
        two.push(arr[1].parse::<i32>().unwrap());
    }
    let mut sum = 0;
    for num in &mut one {
        sum += *num * (two.iter().filter(|&n| *n == *num).count() as i32);
    }
    println!("{}", sum);
}